pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,573 @@
+<script lang="ts">
+import Vue from 'vue';
+
+type Error = 'duplicate' | 'empty';
+
+type Arrow = 'up' | 'down';
+
+const DIRECTION: Record<Arrow, number> = {
+  up:   -1,
+  down: 1,
+};
+
+const INPUT = {
+  create: 'item-create',
+  edit:   'item-edit',
+};
+
+const BOX = 'box';
+
+const CLASS = {
+  item:  'item',
+  error: 'error',
+};
+
+function findIndex(items: string[], item: string, trim = true) {
+  return items.indexOf(trim ? item?.trim() : item);
+}
+
+function findDuplicates(items: string[], caseSensitive = true) {
+  return items.filter((item, idx) => {
+    const index = findIndex(
+      items.map(i => (caseSensitive ? i : i.toLowerCase()).trim()),
+      (caseSensitive ? item : item.toLowerCase()),
+    );
+
+    return idx !== index;
+  }).length > 0;
+}
+
+/**
+ * Manage a list of strings
+ */
+export default Vue.extend({
+  name:  'string-list',
+  props: {
+    /**
+     * The items source
+     */
+    items: {
+      type: Array,
+      default() {
+        return [];
+      },
+    },
+    /**
+     * The caseSensitive option when create or edit items
```

Comment:
It might be a good idea to mention what `caseSensitive` prop controls. It looks like it's used for detecting duplicate items.